frontend/viewer/src/lib/services/idle-service.ts

```diff
@@ -0,0 +1,23 @@
+import {IsIdle} from 'runed';
+import {getContext, hasContext, setContext} from 'svelte';
+
+const symbol = Symbol.for('fw-lite-idle-service');
+const DEFAULT_IDLE_TIMEOUT_MS = 5 * 60 * 1000;
+
+export function useIdleService(timeout?: number): IdleService {
+  if (hasContext(symbol)) return getContext(symbol);
```

Comment:
we should verify this works as we expect. I think right now each input will get it's own instance as none of it's parents put the idle service in the context and the context is scoped to the component hierarchy. We could just put a `useIdleService` call in the app root to solve this.

Reply:
Fixed in commit b60a59bf2.